src/querybuilder/components/Trace/index.tsx

```diff
@@ -17,6 +19,38 @@ interface Props {
 
 export const TraceView = React.memo<Props>(({ query, data, datasource }) => {
   const [trace, setTrace] = useState<Trace | null>(null)
+  const [openModal, setOpenModal] = useState(false)
+  const [copied, setCopied] = useState(false)
+  const styles = useStyles2(getStyles);
+
+  const handleOpenModal = () => {
+    setOpenModal(true)
+  }
+
+  const handleCloseModal = () => {
+    setOpenModal(false)
+  }
+
+  const handleCopyToClipboard = () => {
+    if (!trace || copied) {return}
+    navigator.clipboard.writeText(trace.JSON)
```

Comment:
i think we should check the permission to write 
```
navigator.permissions.query({name: 'clipboard-write'})
```

Reply:
Good point, I will add a try-catch.

Using navigator.permissions.query({name: 'clipboard-write'}) is not the best option because only Chromium support `name: 'clipboard-write'`:
https://developer.mozilla.org/en-US/docs/Web/API/Permissions/query
> The name of the API whose permissions you want to query. Each browser supports a different set of values. You'll find Firefox values [there](https://searchfox.org/mozilla-central/source/dom/webidl/Permissions.webidl#10), Chromium values [there](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/third_party/blink/renderer/modules/permissions/permission_descriptor.idl), and WebKit values [there](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/Modules/permissions/PermissionName.idl).